2025-04-14-January model to send for review.nlogo

```diff
@@ -0,0 +1,1081 @@
+
+  ; calculate B-side of initial odds B:A, with A-side set to 1
+  set odds credence / (1 - credence)
+  ; store prior odds
+  set last-odds odds
+
+
+  ; update the odds with each result one at a time
+  repeat total-positive [                 ; update on each positive result
+    set odds odds * likelihood-positive
+    set pos-updates (pos-updates + 1)
+  ]
+
+  repeat total-negative [       ; update on each negative result
+    set odds odds * likelihood-negative
+    set neg-updates (neg-updates + 1)
+  ]
+    set credence odds / (1 + odds)     ; update credences given new odds
+
+  ; Update the turtle's label to reflect the new credence
+  set label precision credence 3
+end
+
+
+
+
+; THESE ARE THE END AND RESTART PROCESSES
+
+to finish-model
+  record-results
+
+
+  if current-run < total-runs [
+    reset-for-next-model-run
+  ]
+  if current-run >= total-runs [
+    print "All runs completed."
+  ]
+end
+
+to record-results
+  if (all? scientists [credence <= 0.5 ]) [
+    set total-failures total-failures + 1
+  ]
+  if (all? scientists [credence >= .99 ]) [
+    set total-successes total-successes + 1
+  ]
+  if ticks >= 500 [
+    set no-convergence no-convergence + 1
+  ]
+  ;; calculate and record average policymaker credence at the end of the run
+calculate-average-p-credence
+  set running-average-p-credence mean average-p-credences-list
+
+  ;; record time to completion of run
+  set time-to-finish ticks
+  set time-to-finish-list lput time-to-finish time-to-finish-list   ;; add time to finish of current run to the list
+  set average-time-to-finish mean time-to-finish-list
+
+end
+
+to calculate-average-p-credence
+  let total-credence sum [credence] of policymakers   ;; sum the credences
+  let num-policymakers count policymakers              ;; count the policymakers
+  let avg-credence total-credence / num-policymakers    ;; compute the average
+  set average-p-credences-list lput avg-credence average-p-credences-list   ;; add average from current run to the list
+end
+
+to reset-for-next-model-run
+  ; Clear turtles and links
+  ask turtles [ die ]
+  ask links [ die ]
+
+  ; Reset patches to initial state, if necessary
+  ask patches [ set pcolor blue ] ; Example: Resetting patch color
+
+  ; Reinitialize global variables specific to each run
+  set average-credences []
+  set final-credences []
+
+  set current-run current-run + 1
+
+  ; Call a modified setup procedure that doesn't reset run-tracking variables
+  setup-no-reset
+end
+
+
+
+to setup-no-reset
+   ifelse network-type = "cycle" [
+    setup-cycle
+  ] [
+    ifelse network-type = "wheel" [
+      setup-wheel
+    ] [
+      if network-type = "complete" [
+        setup-complete ; Assuming "connected" refers to a fully connected network, and renamed accordingly
+      ]
+    ]
+  ]
+  set average-credences []
+  set final-credences []
+  setup-propagandists
+  setup-policymakers
+  link-prop-to-sci
+  link-prop-to-pol
+end
+@#$#@#$#@
+GRAPHICS-WINDOW
```

Comment:
If you are commited to Netlogo you might consider the value of visual plots, especially given the respect you give to saving priors.